crates/agentgateway/src/http/ext_authz.rs

```diff
@@ -139,32 +168,29 @@ impl ExtAuthz {
 		// https://github.com/envoyproxy/envoy/blob/d9e0412bd471a80e0938102c0c8cbff1caedd4cf/source/common/http/header_map_impl.cc#L28-L33
 		let mut headers = std::collections::HashMap::new();
 
-		let header_allowlist: std::collections::HashSet<&str> = self
-			.include_request_headers
-			.iter()
-			.map(|s| s.as_str())
-			.collect();
-
-		for name in req.headers().keys() {
-			if !self.include_request_headers.is_empty() && !header_allowlist.contains(name.as_str()) {
-				continue;
+		if self.include_request_headers.is_empty() {
+			for name in req.headers().keys() {
+				self.get_header_values(req, name, &mut headers);
 			}
-
-			let values: Vec<String> = req
-				.headers()
-				.get_all(name)
-				.iter()
-				.filter_map(|v| v.to_str().ok())
-				.map(|s| s.to_string())
-				.collect();
-
-			if !values.is_empty() {
-				let joined = if name.as_str() == "cookie" {
-					values.join("; ")
-				} else {
-					values.join(", ")
-				};
-				headers.insert(name.as_str().to_string(), joined);
+		} else {
+			// Only include requested headers (both regular and pseudo headers)
+			for header_spec in &self.include_request_headers {
```

Comment:
nit: could make `include_request_headers` a `HeaderOrPseudo` already instead of `String`